### ISSUE TITLE
Repeated usage of mock fetch with delay

### ISSUE THREAD

**Issue by dpwrussell** [wheresrhys/fetch-mock#468]
Hi,

I just started using your library and I ran into an issue when testing a function where I repeatedly use fetch until success. The `fetchTimeout` function aborts attempts to fetch after a preset period. In this example I abort after 10ms for purposes of demonstration.

```js
async function fetchTimeout(url, options) {
  // Abort a request after 10ms
  const controller = new AbortController();
  const timeout = setTimeout(() => { controller.abort() }, 10);
  const optionsSignal = {
    ...options,
    signal: controller.signal
  };

  try {
    return await fetch(url, optionsSignal);
  } finally {
    clearTimeout(timeout);
  }
}
```

I call the above `fetchTimeout` function until I get a satisfactory response or a limit is reached. Between each call to fetchTimeout I back off for 500ms. With the below setup, I was expecting for the fetch to never succeed because the 5000ms delay I configured for fetch, should always be longer than the 10ms fetch is given until it aborts. In my real code, all of the retry logic (which is more robust and complicated than below) is of course in my source code, not in my test code, but I thought this would be helpful to demonstrate.

```js
      const mockFetch = fetchMock.sandbox().get(
        'http://foo.bar.com',
        200,
        {
          delay: 5000
        }
      );

      sinon.replace(fetch, 'default', mockFetch);

      for (let i = 0; i<100; i++) {
        console.log(`Run ${i}`);
        try {
          const res = await fetchTimeout('http://foo.bar.com', {});
          if (res.ok) {
            console.log('Breaking out!');
            break;
          }
        } catch(err) {
          // Swallow AbortErrors only
          if (err.name !== 'AbortError') {
            throw(err);
          }
        }
        // Delay for 500ms
        await new Promise(resolve => setTimeout(resolve, 500));
      }
```

It looks like the 5000ms `delay` is somehow not per call to fetch however. After a few iterations (roughly 10, which fits with the 500ms backoff), the for loop will abort because fetch succeeeded.

Is this intended? I would have thought that the delay should be per call to fetch.

**Comment by wheresrhys** [comment]
Thanks for the detailed run through. I can totally see why it doesn't work, and will try to publish a fix soon

**Comment by wheresrhys** [comment]
Fixed in 7.7.1

Thanks again for the detailed bug report
